server_spidercast/src/MCC_RUM/src/rmmCutils.c

```diff
@@ -53,18 +53,21 @@ size_t rmm_strllen(const char *src, size_t size)
 size_t rmm_strlcpy(char *dst, const char *src, size_t size)
 {
   size_t rc=0;
-  char *p, *q;
+ //char *p, *q;
   const char *s;
 
   if ( src )
   {
     s = src ;
     if ( dst && size > 0 )
     {
-      p = dst ;
-      q = p + (size - 1) ;
-      while ( p<q && *s ) *p++ = *s++ ;
-      *p=0;
+      size_t len =0; // new variable to track the index in dst
+      while ( len <  (size -1) && *s)  //updated while loop to use len for indexing
+      {
+        dst [len] = *s++;
+        len++;
+      }
+      dst[len] = '\0';
```

Comment:
Can you explain to me why you're making this change? The code looked safe to me before (but p and q were terrible names) - and the comments aren't right - they only make sense during the review - someone looking at this in ten years isn't going to think len is new.

Reply:
The approach for changing this was previously p and q pointers were used  to check the position is dst and I the chnage that I made has variable len to count the number of char and to keep the array in bound. It was made to avoid the pointer arithmetic.

Reply:
Sorry, I should have added comments in brief explaining the change properly.
Should I revert the code back to how it was and go for another approach?

Reply:
I'm not saying revert it :) I don't know yet - is there a warning you're trying to fix? Can you share it here?

Reply:
This is the warning I was trying to resolve:
`[exec] /opt/amlen/server_spidercast/src/MCC_RUM/src/rmmCutils.c:66:22: warning: array subscript 8 is outside array bounds of char[8] [-Warray-bounds]`

Reply:
This change doesn't remove that warning - with the code change it is still there:

```
In file included from /home/jon/work/amlen/repos/amlen/server_spidercast/src/MCC_RUM/src/RumReceiverApi.c:139:
In function ‘rmm_strlcpy’,
    inlined from ‘rmmGetAddrInfo’ at /home/jon/work/amlen/repos/amlen/server_spidercast/src/MCC_RUM/src/SockMgm.c:1109:5,
    inlined from ‘cip_handle_conn_req’ at /home/jon/work/amlen/repos/amlen/server_spidercast/src/MCC_RUM/src/ConnectionHandler.c:1515:10,
    inlined from ‘ConnectionHandler’ at /home/jon/work/amlen/repos/amlen/server_spidercast/src/MCC_RUM/src/ConnectionHandler.c:257:7:
/home/jon/work/amlen/repos/amlen/server_spidercast/src/MCC_RUM/src/rmmCutils.c:64:35: warning: array subscript 8 is outside array bounds of ‘char[8]’ [-Warray-bounds=]
   64 |       while ( len <  (size -1) && *s)  //updated while loop to use len for indexing
```

But I *think* the compiler is wrong - in both the original version and your altered version, it will only go off the send of src if src is not null terminated and in this case, the offending call is i a function:
```
int rmmGetAddrInfo(char *addr0, char *port0, int family, int stype, int proto, int flags, AI *result, int *errCode, char *errMsg)
```
The line in the callstack is
```
 strlcpy(port,port0,sizeof(port)) ;
```
 where the definitions are `char port[16] , *pp ;` and `char *port0`
and the caller of that line is
  
 ```
      if ( rmmGetAddrInfo(cInfo->req_addr,port,af,SOCK_STREAM,IPPROTO_TCP,0,&tmp_ipi.ai,&errCode,errMsg) != RMM_SUCCESS )
```      
      
where:
  char port[8] , errMsg[ERR_MSG_SIZE];
 and port is initialised like:
 snprintf(port,8,"%d",cInfo->req_port) ; 
 
 So I think the compiler thinks src in the strcpy function is not nullterminated but the `snprintf(port,8,"%d",cInfo->req_port) ; `
 
 means it *always* is

Reply:
I think that this particular warning often shows false positives:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56456
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=109071

Is there a way we can disable the warning just for the few lines of code in question (I've not researched that).